expo/scheduler/functions.py

```diff
@@ -436,3 +437,121 @@ def get_roles_from_scheduler(workeritems, conference):
+                  'Show', etc, or a list or tuple of event_types
+    day         - Which day of the conference to get calendar info for,
+                  None or 'All' gets info for entire conference
+    '''
+    #  TODO: Add ability to filter on a users schedule for things like
+    #  volunteer shifts.
+
+    from gbe.functions import get_conference_by_slug
+    import gbe.models as conf
+    from expo.settings import (
+        DATETIME_FORMAT,
+        TIME_FORMAT,
+        SITE_URL,
+        )
+    import csv
+
+    if conference is None:
+        conference = conf.Conference.current_conf()
+    else:
+        conference = get_conference_by_slug(conference)
+
+    if day == 'All':
+        day = None
+    cal_times = cal_times_for_conf(conference, day)
+
+    local_domain = SITE_URL.split('://')[1].replace('www.', '')
+    if event_types == 'All':
+        event_types = ['Show',
+                       'Class',
+                       'Special Event',
+                       'Master Class',
+                       'Drop-In Class']
+    elif event_types == 'Show':
+        event_types == ['Show',
+                        'Special Event',
+                        'Master Class',
+                        'Drop-In Class',
+                        ]
+    elif not type(event_types).isinstance([]) or \
+            type(event_types).isinstance(()):
+        event_types = [event_types]
+    events = []
+    for event_type in event_types:
+        events = events + event_info(confitem_type=event_type,
+                                     cal_times=cal_times,
+                                     conference=conference)
+    if cal_format == 'gbook':
+        return_file = '"Session Title","Date","Time Start","Time End",' + \
+                      '"Room/Location","Schedule Track (Optional)",' + \
+                      '"Description (Optional)",""\r\n'
+
+        for event in events:
+            csv_line = '"%s",' % (event['title'])
+            csv_line = csv_line + '"%s",' % \
+                (date_format(event['start_time'], 'DATE_FORMAT')
+                 .replace(',', ''))
+            csv_line = csv_line + '"%s",' % \
+                (date_format(event['start_time'], 'TIME_FORMAT')
+                 .replace('a.m.', 'AM').replace('p.m.','PM')
+                 .replace('am', 'AM').replace('pm','PM'))
+            csv_line = csv_line + '"%s",' % \
+                (date_format(event['stop_time'], 'TIME_FORMAT')
+                 .replace('a.m.', 'AM').replace('p.m.','PM')
+                 .replace('am', 'AM').replace('pm','PM'))
+            csv_line = csv_line + '"%s",' % (event['location'])
+            csv_line = csv_line + '"%s",' % (event['type'].split('.')[0])
+            csv_line = csv_line + '"%s",' % \
+                (event['description'].replace('\n', '').replace('\r', ''))
+            csv_line = csv_line + '"%s%s"\r\n' % (SITE_URL, event['link'])
+            return_file = return_file + csv_line
```

Comment:
Also, we can't use "noon" - anything that starts or ends at "noon" instead of 12:00 PM will cause the import to guidebook to fail.. this thing is wicked fussy.